pymc3/ode/ode.py

```diff
@@ -1,211 +1,222 @@
-        states, sens = super(theano.Op, self).__call__(y0, theta, **kwargs)
-
-        if theano.config.compute_test_value != 'off':
-            # compute test values from input test values
-            test_states, test_sens = self._simulate(
-                y0=self._get_test_value(y0),
-                theta=self._get_test_value(theta)
-            )
-
-            # check types of simulation result
-            if not test_states.dtype == self._otypes[0].dtype:
-                raise TypeError('Simulated states have the wrong type')
-            if not test_sens.dtype == self._otypes[1].dtype:
-                raise TypeError('Simulated sensitivities have the wrong type')
-
-            # check shapes of simulation result
-            expected_states_shape = (self.n_times, self.n_states)
-            expected_sens_shape = (self.n_times, self.n_states, self.n_p)
-            if not test_states.shape == expected_states_shape:
-                raise ShapeError('Simulated states have the wrong shape.', test_states.shape, expected_states_shape)
-            if not test_sens.shape == expected_sens_shape:
-                raise ShapeError('Simulated sensitivities have the wrong shape.', test_sens.shape, expected_sens_shape)
-
-            # attach results as test values to the outputs
-            states.tag.test_value = test_states
-            sens.tag.test_value = test_sens
-        
-        if return_sens:
-            return states, sens
-        return states
-
-    def perform(self, node, inputs_storage, output_storage):
-        y0, theta = inputs_storage[0], inputs_storage[1]
-        # simulate states and sensitivities in one forward pass
-        output_storage[0][0], output_storage[1][0] = self._simulate(y0, theta)
-
-    def infer_shape(self, node, input_shapes):
-        s_y0, s_theta = input_shapes
-        output_shapes = [(self.n_times, self.n_states), (self.n_times, self.n_states, self.n_p)]
-        return output_shapes
-
-    def grad(self, inputs, output_grads):
-        _log.debug('grad w.r.t. inputs {}'.format(hash(tuple(inputs))))
-        
-        # fetch symbolic sensitivity output node from cache
-        ihash = hash(tuple(inputs))
-        if ihash in self._output_sensitivities:
-            sens = self._output_sensitivities[ihash]
-        else:
-            _log.debug('No cached sensitivities found!')
-            _, sens = self.__call__(*inputs, return_sens=True)
-        ograds = output_grads[0]
-        
-        # for each parameter, multiply sensitivities with the output gradient and sum the result
-        # sens is (n_times, n_states, n_p)
-        # ograds is (n_times, n_states)
-        grads = [
-            tt.sum(sens[:,:,p] * ograds)
-            for p in range(self.n_p)
-        ]
-        
-        # return separate gradient tensors for y0 and theta inputs
-        result = tt.stack(grads[:self.n_states]), tt.stack(grads[self.n_states:])
-        return result
+import logging
```

Comment:
I wonder why this diff is so bad at tracking what actually changed

Reply:
Either it's something with the encoding, or because there's a change every few lines (comments and dtypes, mostly). VS did show the diff though.